lms/static/coffee/src/instructor_dashboard/send_email.coffee

```diff
@@ -15,147 +15,154 @@ create_email_content_table = -> window.InstructorDashboard.util.create_email_con
-    $(".msg-confirm").css({"display":"block"})
+class @SendEmail
+    constructor: (@$container) ->
+        # gather elements
+        @$emailEditor = XBlock.initializeBlock($('.xblock-studio_view'))
+        @$send_to = @$container.find("input[name='send_to']")
+        @$subject = @$container.find("input[name='subject']")
+        @$btn_send = @$container.find("input[name='send']")
+        @$task_response = @$container.find(".request-response")
+        @$request_response_error = @$container.find(".request-response-error")
+        @$content_request_response_error = @$container.find(".content-request-response-error")
+        @$history_request_response_error = @$container.find(".history-request-response-error")
+        @$btn_task_history_email = @$container.find("input[name='task-history-email']'")
+        @$btn_task_history_email_content = @$container.find("input[name='task-history-email-content']'")
+        @$table_task_history_email = @$container.find(".task-history-email-table")
+        @$table_email_content_history = @$container.find(".content-history-email-table")
+        @$email_content_table_inner = @$container.find(".content-history-table-inner")
+        @$email_messages_wrapper = @$container.find(".email-messages-wrapper")
+
+        # attach click handlers
+
+        @$btn_send.click =>
+            subject = @$subject.val()
+            body = @$emailEditor.save()['data']
+            targets = []
+            @$send_to.filter(':checked').each ->
+                targets.push(this.value)
+
+            if subject == ""
+                alert gettext("Your message must have a subject.")
+
+            else if body == ""
+                alert gettext("Your message cannot be blank.")
+
+            else if targets.length == 0
+                alert gettext("Your message must have at least one target.")
+
+            else
+                # Validation for keyword substitution
+                validation = KeywordValidator().validate_string body
+                if not validation.is_valid
+                    message = gettext("There are invalid keywords in your email. Please check the following keywords and try again:")
+                    message += "\n" + validation.invalid_keywords.join('\n')
+                    alert message
+                    return
+
+                target_map = {
+                        "myself": gettext("Yourself"),
+                        "staff": gettext("Everyone who is staff or instructor on this course"),
+                        "learners": gettext("All learners enrolled in this course"),
+                }
+                success_message = gettext("Your email was successfully queued for sending. Please note that for large classes, it may take up to an hour (or more, if other courses are simultaneously sending email) to send all emails.")
+                confirm_message = gettext("You are about to send an email titled '<%= subject %>' to the following targets. Is this OK?")
```

Comment:
"You are sending an email message with the subject ```<%=subject %>``` to the following recipients. Is this OK?"